### DISCUSSION TITLE
How to use api token in some enpoints

### DISCUSSION THREAD

**Discussion by enieber** [loco-rs/loco#319] [Q&A] [answered]
It's possiblem use api token in some requests? sample:

// use api token in this request to create product
POST: /products

**Comment by kaplanelad** [comment, marked as answer]
Hey, @enieber, yes, this is something that Loco provides already. 

go you your controller and add `auth: auth::ApiToken<users::Model>` middleware into your controller. 
for example: 
```rust
async fn products(
    auth: auth::ApiToken<users::Model>,
    State(_ctx): State<AppContext>,
) -> Result<Json<CurrentResponse>> {
   ...
}
```

The `ApiToken` will check the user token. 

It looks like we didn't document it at all. Nice catch! 
I have opened an issue for that https://github.com/loco-rs/loco/issues/325
